wdl-analysis/src/scope/v1.rs

```diff
@@ -0,0 +1,876 @@
+    ) {
+        for decl in section.declarations() {
+            let name = decl.name();
+            let context = ScopedNameContext::Output(name.span());
+            if let Some(prev) = names.get(name.as_str()) {
+                diagnostics.push(name_conflict(
+                    name.as_str(),
+                    context.into(),
+                    prev.context().into(),
+                ));
+                continue;
+            }
+
+            names.insert(
+                name.as_str().to_string(),
+                ScopedName::new(context, decl.syntax().green().into(), false),
+            );
+        }
+    }
+
+    /// Adds a task scope to the document's scope.
+    fn add_task_scope_v1(&mut self, task: &v1::TaskDefinition, diagnostics: &mut Vec<Diagnostic>) {
+        // Check for a conflict with another task or workflow
+        let name = task.name();
+        if let Some(s) = self.tasks.get(name.as_str()) {
+            diagnostics.push(name_conflict(
+                name.as_str(),
+                NameContext::Task(name.span()),
+                NameContext::Task(s.name_span),
+            ));
+            return;
+        } else if let Some(s) = &self.workflow {
+            if s.name == name.as_str() {
+                diagnostics.push(name_conflict(
+                    name.as_str(),
+                    NameContext::Task(name.span()),
+                    NameContext::Workflow(s.name_span),
+                ));
+                return;
+            }
+        }
+
+        // Populate the scope's names
+        let mut names: IndexMap<_, ScopedName> = IndexMap::new();
+        let mut saw_input = false;
+        let mut saw_output = false;
+        for item in task.items() {
+            match item {
+                v1::TaskItem::Input(section) if !saw_input => {
+                    saw_input = true;
+                    Self::add_inputs(&mut names, &section, diagnostics);
+                }
+                v1::TaskItem::Output(section) if !saw_output => {
+                    saw_output = true;
+                    Self::add_outputs_v1(&mut names, &section, diagnostics);
+                }
+                v1::TaskItem::Declaration(decl) => {
+                    let name = decl.name();
+                    let context = ScopedNameContext::Decl(name.span());
+                    if let Some(prev) = names.get(name.as_str()) {
+                        diagnostics.push(name_conflict(
+                            name.as_str(),
+                            context.into(),
+                            prev.context().into(),
+                        ));
+                        continue;
+                    }
+
+                    names.insert(
+                        name.as_str().to_string(),
+                        ScopedName::new(context, decl.syntax().green().into(), false),
+                    );
+                }
+                v1::TaskItem::Input(_)
+                | v1::TaskItem::Output(_)
```

Comment:
The first ones have conditional matches, so these are when that condition isn't true.

There's no particular reason for not using a wildcard match, but for enum variants that aren't that numerous, I like to use exhaustive matches so that any newly added items will cause errors to indicate changes are required here.

Reply:
Got it. Thanks for the explanations!